modules/host/host.go

```diff
@@ -148,6 +158,38 @@ func (h *Host) Capacity() int64 {
 	return h.spaceRemaining
 }
 
+// Close shuts down the host, preparing it for garbage collection.
+func (h *Host) Close() error {
+	// The order in which things are closed has been explicitly chosen to
+	// minimize turbulence in the event of an error.
+
+	// Save the latest host state.
+	err := func() error {
+		h.mu.Lock()
+		defer h.mu.Unlock()
+		return h.save()
+	}()
```

Comment:
It's more than one line of code under lock, just deceptive. The alternative:

```
h.mu.Lock()
err := h.save()
if err != nil {
    h.mu.Unlock() // prone to be forgotten! pattern common in project history!
    return err
}
h.mu.Unlock()
```

As you can see, that's actually 5 lines of code under lock, and it includes a pattern that has caused frustrating deadlocks in the past. The way I've done it is more verbose by only 1 line of code, but explicitly avoids a known problematic pattern.

I'm very strict with the locking stuff because it's easy to make a mistake and very difficult to debug even when the mistake is a harmless as forgetting to unlock before returning, especially if you forgot when returning an error, because that means you have to figure out that the deadlock is only repeatable in the event of the error.  We've lost several days at a time previously to locking mistakes and I'm eager to avoid that in the future.

I do want to make it a convention that you need an anonymous function anytime you do locking that would require making more than 1 call to unlock (most common example is to return an error), or any time that the lock is held for more than 5 lines.

oh, I guess you could:

```
h.mu.Lock()
err := save()
h.mu.Unlock()
if err != nil {
    return err
}
```

But I'm strongly opposed to anything that would require writing multiple calls to unlock for a single call of lock (due to branching), **strongly** in favor of clearing indicating the scope in which a lock is held.